gputop/remote/gputop.js

```diff
@@ -131,27 +131,33 @@ function Counter () {
     this.samples_ = 0; // Number of samples processed
     this.updates = [];
     this.graph_data = [];
+    this.graph_options = []; /* each counter has its own graph options so that
+                              * we can adjust the Y axis for each of them */
     this.units = '';
     this.graph_markings = [];
     this.record_data = false;
     this.eq_xml = ""; // mathml equation
     this.max_eq_xml = ""; // mathml max equation
-
+    this.duration_dependent = true;
     this.test_mode = false;
 }
 
 Counter.prototype.append_counter_data = function (start_timestamp, end_timestamp,
                                                   d_value, max, reason) {
-     if (this.record_data && max != 0) {
-        this.updates.push([start_timestamp, end_timestamp, d_value, max, reason]);
 
+    if (this.duration_dependent && this.latest_duration) {
+        var per_sec_scale = 1000000000 / this.latest_duration;
```

Comment:
At this point the duration we should use should be calculated from `end_timestamp - start_timestamp`

`this.latest_duration` is calculated in the same way later in this function (so the first time this function is called then `latest_duration` won't be initialized or will be out-of-date).